task2.c

```diff
@@ -0,0 +1,94 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <time.h>
+
+void swap(int *firstElement, int *secondElement)
+{
+    int transitElement = *firstElement;
+    *firstElement = *secondElement;
+    *secondElement = transitElement;
+}
+
+void printArray(int *array, int size)
+{
+    for (int i = 0; i < size; ++i)
+    {
+        printf("%d ", array[i]);
+    }
+    printf("\n");
+}
+
+bool isSorted(int *array, int arrayLength)
+{
+    for (int i = 0; i < arrayLength - 1; ++i)
+    {
+        if (array[i] > array[i + 1])
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+void shuffle(int *array, int arrayLength)
+{
+    srand(time(0));
+    for (int i = 0; i < arrayLength; ++i)
+    {
+        swap(&array[i], &array[rand() % arrayLength]);
+    }
+}
+
+void bogosort(int *array, int arrayLength)
+{
+    int iterationCounter = 0;
+    while (!isSorted(array, arrayLength))
+    {
+        if (iterationCounter % 10000000 == 0)
+        {
+            printf("Processing... :)\n");
+        }
+        shuffle(array, arrayLength);
+        ++iterationCounter;
+    }
+}
+
+int main(void)
```

Comment:
```suggestion
int main()
```
Это осмысленно писать только для сигнатур функций.